spec.bs

```diff
@@ -3003,6 +3014,49 @@ The <dfn for=Navigator method>getInterestGroupAdAuctionData(|config|)</dfn> meth
 
 </div>
 
+<div algorithm>
+  To <dfn>parse and verify ad auction config</dfn> given an {{AdAuctionDataConfig}} |configIDL|:
+
+  1. Let |config| be a new [=auction data config=] with the following [=struct/items=]:
+    : [=auction data config/seller=]
+    :: The result of running [=parsing an https origin=] on
+       |configIDL|["{{AdAuctionDataConfig/seller}}"]
+    : [=auction data config/coordinator=]
+    :: The result of running [=parsing an https origin=] on
+       |configIDL|["{{AdAuctionDataConfig/coordinatorOrigin}}"]
+    : [=auction data config/request size=]
+    :: |configIDL|["{{AdAuctionDataConfig/requestSize}}"]
+    : [=auction data config/per buyer config=]
+    :: The result of running [=parse per buyer auction data configs=] on
+      |configIDL|["{{AdAuctionDataConfig/perBuyerConfig}}"]
+  1. If |config|'s [=auction data config/seller=] or [=auction data config/coordinator=]
+     fields are failure, then [=exception/throw=] a {{TypeError}}.
+  1. If |config|'s [=auction data config/per buyer config=] [=map/is not empty=]
+     and |config|'s [=auction data config/request size=] is null:
+     1. Let |request_size| equal 0.
+     1. [=list/For each] |buyerConfig| of |config|'s [=auction data config/per buyer config=]'s [=map/values=]:
+        1. If |buyerConfig|'s [=auction data buyer config/size=] is null,
+           then [=exception/throw=] a {{TypeError}}.
+        1. Set |request_size| to |request_size| + |buyerConfig|'s [=auction data buyer config/size=]
+    1. Set |config|'s [=auction data config/request size=] to |request_size|.
+  1. Return |config|.
+</div>
+
+<div algorithm>
+  To <dfn>parse per buyer auction data configs</dfn> given an
+  {{AdAuctionDataBuyerConfig}} |perBuyerConfigIDL|:
+  1. Let |configs| be a new [=ordered map=] whose [=map/keys=] are [=origins=]
+     and whose [=map/values=] are [=auction data config=]
+  1. [=map/For each=] |buyerIDL| → |buyerConfigIDL| of |perBuyerConfigIDL|:
+    1. Let |buyerConfig| be a new [=auction data buyer config=] with the following [=struct/items=]:

```

Comment:
nit: since this only has one field, it can be "a new [=...=] whose [=auction data buyer config/size=] is ...". But it's fine to keep it as is if it's possibly to be extended.

Reply:
Done.